### PR TITLE
Ignore console-log.txt when checking for empty workspace in e2e.sh

### LINKED ISSUE #18441: Cluster didn't started correctly because of non-empty workspace
I've already seen it a couple of time.

The cluster just don't start with the following logs:

    06:31:54 Pulling binaries from GCS
    06:31:54 + [[ '' =~ ^[yY]$ ]]
    06:31:54 ++ find .
    06:31:54 ++ wc -l
    06:31:54 + [[ 2 != 1 ]]
    06:31:54 + echo /jenkins-master-data/jobs/kubernetes-e2e-gce-parallel/workspace not empty, 'bailing!'
    06:31:54 /jenkins-master-data/jobs/kubernetes-e2e-gce-parallel/workspace not empty, bailing!
    06:31:54 + exit 1

The example failure:
http://kubekins.dls.corp.google.com/view/Critical%20Builds/job/kubernetes-e2e-gce-parallel/12586/console

@kubernetes/goog-testing

### PR DESCRIPTION
**Pull request by spxtr**
Should fix #18441. It might be worth adding a white list if this gets any longer.

cc @ixdy

### CODE REVIEW

**Comment by wojtek-t** [comment]
LGTM

**Comment by wojtek-t** [comment]
@k8s-bot unit test this please

**Comment by spxtr** [comment]
PR Jenkins having trouble again?

**Comment by wojtek-t** [comment]
@k8s-bot e2e test this please

File: hack/jenkins/e2e.sh [hunk 1]
```diff
@@ -1145,8 +1145,9 @@ if [[ "${E2E_UP,,}" == "true" || "${JENKINS_FORCE_GET_TARS:-}" =~ ^[yY]$ ]]; the
         # Otherwise, we want a completely empty directory.
         if [[ "${JENKINS_FORCE_GET_TARS:-}" =~ ^[yY]$ ]]; then
             rm -rf kubernetes*
-        elif [[ $(find . -not -path './.config*' | wc -l) != 1 ]]; then
-            echo $PWD not empty, bailing!
+        elif [[ $(find . -not -path "./.config*" -not -name "console-log.txt" \
```
Comment (jlowdermilk) [inline hack/jenkins/e2e.sh]: Sorry to request change after lgtm, but can you add a short comment here explaining why we expect these files to show up in a "clean" workspace?

Comment (spxtr) [inline hack/jenkins/e2e.sh]: Done, thanks.

**Comment by jlowdermilk** [comment]
PR builder is currently broken, so not waiting for results

**Comment by jlowdermilk** [comment]
@spxtr can you cherrypick this onto 1.1 as well? At least one job is hitting this.

**Comment by spxtr** [comment]
@jlowdermilk Okay I think I got 1.1 right. Should I do 1.0 as well?

**Comment by jlowdermilk** [comment]
Yes please.

**Comment by ixdy** [comment]
Hm, this fix seems wrong. console-log.txt is not created by Jenkins, it's created by `hack/jenkins/upload-to-gcs.sh`, but it shouldn't be created before the end of a test run - something is fishy here.

**Comment by spxtr** [comment]
Is it possible that upload_logs_to_gcs is firing after the next build starts?

**Comment by ixdy** [comment]
ah, that's what appears to have happened. @wojtek-t aborted build 3441, and it seems the upload_logs_to_gcs function didn't finish running before Jenkins started the next run (which then failed, because console-log.txt existed). Kinda weird that Jenkins didn't wait for everything to exit, but maybe the `source` hack we're doing has that effect?

**Comment by spxtr** [comment]
signals/traps are asynchronous, so as far as Jenkins is concerned, it's finished as soon as e2e.sh returns, before upload_logs_to_gcs does its thing.

**Comment by spxtr** [comment]
Looks like it also occurred in disruptive-1.1, which wasn't aborted beforehand. It did fail before, though.

**Comment by ixdy** [comment]
The previous build of disruptive-1.1 aborted due to timeout, so it's pretty likely that it leaked through again.

I think a better option is to make sure `upload_logs_to_gcs` doesn't run past the end of a test run. I'll look into getting that set up.

**Comment by ixdy** [comment]
Planning to use https://wiki.jenkins-ci.org/display/JENKINS/PostBuildScript+Plugin and eliminate the gross `source` hack. If that works, we should be able to revert this PR.

**Pull request merged**